dynet/expr.cc

```diff
@@ -44,15 +43,17 @@ Expression operator+(const Expression& x, const Expression& y) {
     else if (y.dim().batch_size() == 1)
         return Expression(x.pg, x.pg->add_function<ScalarAdd>({x.i, y.i}));
     else
-        return Expression(x.pg, x.pg->add_function<Sum>({x.i, y.i}));
+        return Expression(x.pg, x.pg->add_function<Sum>({x.i, y.i}, x.pg->nodes[x.i]->device));
```

Comment:
This shouldn't need a "device" argument, right? It can just inherit its device from its inputs.